### PR TITLE
Read offloaded literals

### PR DESCRIPTION
**Pull request by eapolinario**
## Tracking issue
Linking to the RFC: https://github.com/flyteorg/flyte/pull/5103

## Why are the changes needed?
As discussed in https://github.com/flyteorg/flyte/pull/5103/, flytekit needs to be able to read offloaded literals.

## What changes were proposed in this pull request?
The type engine "unrolls" literals that contain offloaded literals prior to turning those into python values.

## Check all the applicable boxes 

- [ ] I updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [ ] All commits are signed-off.

### CODE REVIEW

**Review by pmahindrakar-oss — DISMISSED** [review]

File: flytekit/models/literals.py [hunk 1]
```diff
@@ -852,6 +852,52 @@ def from_flyte_idl(cls, pb2_object):
         )
 
 
+class LiteralOffloadedMetadata(_common.FlyteIdlEntity):
```
Comment (pmahindrakar-oss) [inline flytekit/models/literals.py:855]: Not your change, but a generator class for this could be useful

**Review by pmahindrakar-oss — APPROVED** [review]

File: flytekit/models/literals.py (hunk 1 above)
Comment (eapolinario) [reply flytekit/models/literals.py:855]: do you mean like a utility function to help in tests?

File: tests/flytekit/unit/core/test_offloaded_literals.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    @task
+    def t1(a: int) -> str:
+        return str(a)
+
+    original_input_literal = literal_models.Literal(
+        scalar=literal_models.Scalar(primitive=literal_models.Primitive(integer=3))
+    )
+
+    # Write offloaded_lv as bytes to a temp file
+    with open(f"{tmp_path}/offloaded_proto.pb", "wb") as f:
```
Comment (thomasjpfan) [inline tests/flytekit/unit/core/test_offloaded_literals.py:28]: I'm surprised using `/` works with windows. I usually go with:

```suggestion
    with (tmp_path / "offloaded_proto.pb").open("wb") as f:
```

**Review by thomasjpfan — APPROVED** [review]
Minor nit on test, otherwise LGTM

File: tests/flytekit/unit/core/test_offloaded_literals.py (hunk 2 above)
Comment (eapolinario) [reply tests/flytekit/unit/core/test_offloaded_literals.py:28]: that's a good call. I'm going to fix this in a separate PR. 

As for what happens on windows, my guess is that the `/` becomes part of the file name.

**Pull request merged**